layers/+misc/all-the-icons/README.org

```diff
@@ -0,0 +1,73 @@
+#+TITLE: All The Icons layer
+
+#+TAGS: misc|layer
+
+* Table of Contents                                       :TOC_5_gh:noexport:
+- [[#description][Description]]
+  - [[#features][Features:]]
+- [[#install][Install]]
+- [[#usage][Usage]]
+
+* Description
+This layer installs and configures All The Icons.
+
+** Features:
+- Enables and configures icons for dired, ivy, and ibuffer when the appropriate layers are enabled.
+- Defines icon transformers for ivy-rich.
+- Enables icons in company-box.
+
+* Install
+To use this configuration layer, add it to your =~/.spacemacs=. You will need to
+add =all-the-icons= to the existing =dotspacemacs-configuration-layers= list in
+this file.
+
+* Usage
+Most of the features provided by this layer work out of the box. However, you'll
+need to configure =ivy-rich= yourself.
+
+This layer provides two transformers for customizing =ivy-rich=:
+=ivy-rich-buffer-icon= and =ivy-rich-file-icon=. They can be used to iconize
+your ivy-rich results as follows.
+
+#+BEGIN_SRC elisp
+  (setq ivy-rich-display-transformers-list
```

Comment:
I'd like to configure this by default, but there isn't a clean way to edit this list without messing with the user's customizations, as far as I know.

Reply:
I would agree, changing this in an automatic way like walking the tree and filling in functions which have not been set could have very suprising effects for many users. Lets just keep it easy we can later on optimise this point in separate PRs.